src/python/zquantum/qaoa/estimators/_cvar_estimator.py

```diff
@@ -68,11 +96,55 @@ def _calculate_expectation_value_for_distribution(
     # Calculates expectation value per bitstring
     expectation_values_per_bitstring = {}
     for bitstring in distribution.distribution_dict:
-        expected_value = Measurements([bitstring]).get_expectation_values(
-            operator, use_bessel_correction=False
-        )
-        expectation_values_per_bitstring[bitstring] = np.sum(expected_value.values)
+        expected_value = _calculate_expectation_value_of_bitstring(bitstring, operator)
+        expectation_values_per_bitstring[bitstring] = expected_value
+
+    return _sum_expectation_values(
+        expectation_values_per_bitstring, distribution.distribution_dict, alpha
+    )
+
+
+def _calculate_expectation_value_for_wavefunction(
+    wavefunction: Wavefunction, operator: IsingOperator, alpha: float
+) -> float:
+    expectation_values_per_bitstring = {}
+    probability_per_bitstring = {}
+
+    n_qubits = wavefunction.amplitudes.shape[0].bit_length() - 1
```

Comment:
I am not familiar with wavefunction.amplitudes. Please make sure that the -1 is needed.